### PR TITLE
Integrate libp2p-ping

### PR DESCRIPTION
**Pull request by xicombd**
Depends on https://github.com/libp2p/js-libp2p-ping/pull/7

### CODE REVIEW

**Comment by diasdavid** [comment]
CI is not happy :(

**Comment by diasdavid** [comment]
@xicombd could you take a look and match it with the migration to pull streams (already done)

**Comment by xicombd** [comment]
@diasdavid build is now ✅

File: test/libp2p.spec.js [hunk 1]
```diff
@@ -415,6 +415,15 @@ describe('libp2p-ipfs', () => {
...
   })
 
+  it('nodeE ping to nodeF', (done) => {
+    const p = nodeE.ping(nodeF.peerInfo)
+
+    p.on('ping', (time) => {
+      p.stop()
+      done()
+    })
+  })
```
Comment (diasdavid) [inline test/libp2p.spec.js]: It does only send one ping? Something tells me this event will be fired up a bunch of times and the only thing preventing the test from failing is that it is the last test, so it closes soon enough

**Comment by diasdavid** [comment]
Sweet! @xicombd :D

**Comment by diasdavid** [comment]
Things to do before finishing:

- [ ] rebase master onto this branch
- [ ] ensure that the test is well handled : https://github.com/ipfs/js-libp2p-ipfs/pull/13#discussion_r78297327
- [ ] make a parallel PR to https://github.com/ipfs/js-libp2p-ipfs-browser, adding the same feature and testing Node.js <-> Browser interop

**Comment by diasdavid** [comment]
looking good @xicombd , just missing the same PR in libp2p-ipfs-browser and we are golden :D

**Comment by diasdavid** [comment]
closing in favour of https://github.com/ipfs/js-libp2p-ipfs-nodejs/pull/42 now with a libp2p-ping fixed and complete and integrated at js-libp2p.

@xicombd thank you for having started this work :)

**Pull request closed without merging**